### PR TITLE
[RSDK-7442] Update deps with moderate severity security issues

### PR DESCRIPTION
**Pull request by hexbabe**
https://github.com/viamrobotics/viam-camera-realsense/pull/33
https://github.com/viamrobotics/viam-camera-realsense/pull/32
https://github.com/viamrobotics/viam-camera-realsense/pull/21

Dependabot has marked multiple dependencies in our Go integration test module as having "moderate severity" security issues. This consolidates all the suggested changes into one PR. Verified that the integration tests still work on linux/arm64

```
rawr-ubuntu@rawr-ubuntu:~/Projects/viam-camera-realsense$ /home/rawr-ubuntu/Projects/viam-camera-realsense/packaging/appimages/deploy/realsense-integration-tests-latest-aarch64.AppImage -module /home/rawr-ubuntu/Projects/viam-camera-realsense/packaging/appimages/deploy/viam-camera-realsense-latest-aarch64.AppImage
VIAM REALSENSE MODULE INTEGRATION TESTS
Tests defined at https://github.com/viamrobotics/viam-camera-realsense/tree/main/integration/tests
checking if file exists at "/home/rawr-ubuntu/Projects/viam-camera-realsense/packaging/appimages/deploy/viam-camera-realsense-latest-aarch64.AppImage"
Starting the tests...
PASS
all tests succeeded!
```

### CODE REVIEW

File: integration/go.mod [hunk 1]
```diff
@@ -73,10 +73,10 @@ require (
 	github.com/klauspost/compress v1.16.5 // indirect
 	github.com/kylelemons/go-gypsy v1.0.0 // indirect
 	github.com/lestrrat-go/backoff/v2 v2.0.8 // indirect
-	github.com/lestrrat-go/blackmagic v1.0.1 // indirect
+	github.com/lestrrat-go/blackmagic v1.0.2 // indirect
```
Comment (nicksanford) [inline integration/go.mod:76]: Are these packages only used in tests?

Comment (hexbabe) [reply integration/go.mod:76]: yep

**Comment by nicksanford** [comment]
@hexbabe Please confirm tests still pass on all other supported platforms. Once you do you can merge it

**Review by nicksanford — APPROVED** [review]

**Comment by hexbabe** [comment]
> @hexbabe Please confirm tests still pass on all other supported platforms. Once you do you can merge it

Docker for Mac worked for building a linux/amd64 appimage for these integration tests! Tested the amd64 appimage works on the UP board

```viam@viam:~$ ./realsense-integration-tests-latest-x86_64.AppImage  -module ./viam-camera-realsense-latest-x86_64.AppImage 
VIAM REALSENSE MODULE INTEGRATION TESTS
Tests defined at https://github.com/viamrobotics/viam-camera-realsense/tree/main/integration/tests
checking if file exists at "./viam-camera-realsense-latest-x86_64.AppImage"
Starting the tests...
PASS
all tests succeeded!```

**Pull request merged**